cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java

```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
+        } catch (AssertionError e) {
+            System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
+                    + "- FAIL - 'collection_prefix=" + collectionPrefix
+                    + "' wrongly encoded as '" + expectedCollectionPrefix + "'");
+            throw e;
+        } // try catch // try catch
+    } // testConfigureCollectionPrefixIsEncodedNewEncoding
     
     /**
      * [NGSIMongoSink.configure] -------- Configured 'db_prefix' is encoded when having forbiden characters.
      */
     @Test
-    public void testConfigureDBPrefixIsEncoded() {
+    public void testConfigureDBPrefixIsEncodedOldEncoding() {
         System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
                 + "-------- Configured 'db_prefix' is encoded when having forbiden characters");
         String collectionPrefix = "sth_";
         String dbPrefix = "this\\is/a$prefix.with forbiden\"chars:-,";
         String dataModel = null; // defaulting
+        String enableEncoding = "false";
         NGSIMongoBaseSinkImpl sink = new NGSIMongoBaseSinkImpl();
-        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel));
-        String encodedDbPrefix = NGSIUtils.encodeSTHDB(dbPrefix);
+        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel,
+                enableEncoding));
+        String expectedDbPrefix = NGSIUtils.encodeSTHDB(dbPrefix);
         
         try {
-            assertTrue(sink.dbPrefix.equals(encodedDbPrefix));
+            assertEquals(expectedDbPrefix, sink.dbPrefix);
             System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
-                    + "-  OK  - 'db_prefix=" + dbPrefix + "' correctly encoded as '" + encodedDbPrefix + "'");
+                    + "-  OK  - 'db_prefix=" + dbPrefix + "' correctly encoded as '" + expectedDbPrefix + "'");
         } catch (AssertionError e) {
             System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
-                    + "- FAIL - 'db_prefix=" + dbPrefix + "' wrongly encoded as '" + encodedDbPrefix + "'");
+                    + "- FAIL - 'db_prefix=" + dbPrefix + "' wrongly encoded as '" + expectedDbPrefix + "'");
             throw e;
-        } // try catch
-    } // testConfigureDBPrefixIsEncoded
+        } // try catch // try catch
+    } // testConfigureDBPrefixIsEncodedOldEncoding
+    
+    /**
+     * [NGSIMongoSink.configure] -------- Configured 'db_prefix' is encoded when having forbiden characters.
+     */
+    @Test
+    public void testConfigureDBPrefixIsEncodedNewEncoding() {
+        System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
+                + "-------- Configured 'db_prefix' is encoded when having forbiden characters");
+        String collectionPrefix = "sth_";
+        String dbPrefix = "this\\is/a$prefix.with forbiden\"chars:-,";
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4